bundler/lib/bundler/cli/init.rb

```diff
@@ -27,25 +28,28 @@ def run
 
         spec = Bundler.load_gemspec_uncached(gemspec)
 
-        File.open(gemfile, "wb") do |file|
+        File.open(@gemfile, "wb") do |file|
           file << "# Generated from #{gemspec}\n"
           file << spec.to_gemfile
         end
       else
-        File.open(File.expand_path("../templates/#{gemfile}", __dir__), "r") do |template|
-          File.open(gemfile, "wb") do |destination|
+        File.open(File.expand_path(template_path, __dir__), "r") do |template|
+          File.open(@gemfile, "wb") do |destination|
             IO.copy_stream(template, destination)
           end
         end
       end
 
-      puts "Writing new #{gemfile} to #{SharedHelpers.pwd}/#{gemfile}"
+      puts "Writing new #{@gemfile} to #{SharedHelpers.pwd}/#{@gemfile}"
     end
 
     private
 
-    def gemfile
-      @gemfile ||= Bundler.preferred_gemfile_name
```

Comment:
Any reason for moving this to `initialize` and change every reference from using an accessor to using the instance variable directly?

Reply:
This was needed in the last approach, I just forgot to remove 😅

I'll remove right now.